pythran/pythonic/numpy/unravel_index.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef PYTHONIC_NUMPY_UNRAVEL_INDEX_HPP
+#define PYTHONIC_NUMPY_UNRAVEL_INDEX_HPP
+
+#include "pythonic/include/numpy/unravel_index.hpp"
+#include "pythonic/__builtin__/ValueError.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace
+  {
+    template <class E, class ShapeIt, class RetIt>
+    void _unravel_index(E expr, ShapeIt shape_it, ShapeIt end_it, RetIt ret_it)
+    {
+      while (shape_it != end_it) {
+        auto &v = *shape_it;
+        *ret_it = expr % v;
+        expr /= v;
```

Comment:
Did you benchmark your implementation vs. numpy version? this modulo + divide looks very costly to me.